Calibration/HcalCalibAlgos/plugins/RecAnalyzerMinbias.cc

```diff
@@ -199,48 +210,54 @@ void RecAnalyzerMinbias::beginRun(edm::Run const&, edm::EventSetup const& iS) {
 	char  name[700], title[700];
 	// For HB
 	int maxDepthHB = hcaltopology->maxDepthHB();
-	int nbinHB     = int(200*eHighHB_);
+	int nbinHB     = (Noise_) ? 18 : int(2000*eHighHB_);
+	double x_min   = (Noise_) ? -3.   : 0.;
+	double x_max   = (Noise_) ? 3.    :  2.*eHighHB_;
 	for (int eta = -50; eta < 50; eta++) {
 	  for (int phi = 0; phi < 100; phi++) {
-	    for (int depth = 1; depth < maxDepthHB; depth++) {
+	    for (int depth = 1; depth <= maxDepthHB; depth++) {
 	      HcalDetId cell (HcalBarrel, eta, phi, depth);
 	      if (hcaltopology->valid(cell)) {
 		sprintf (name, "HBeta%dphi%ddep%d", eta, phi, depth);
-		sprintf (title,"Energy (HB #eta %d #phi %d depth %d)", eta, phi, depth);
-		TH1D* h = new TH1D(name, title, nbinHB, 0, 2*eHighHB_);
-		histHB_[cell] = h;
+		sprintf (title,"HB #eta %d #phi %d depth %d", eta, phi, depth);
+		TH1D* h = fs_->make<TH1D>(name, title, nbinHB, x_min, x_max);  
+		histHC_[cell] = h;
 	      }
 	    }
 	  }
 	}
 	// For HE
-	int maxDepthHE = hcaltopology->maxDepthHB();
-	int nbinHE     = int(200*eHighHE_);
+	int maxDepthHE = hcaltopology->maxDepthHE();
+	int nbinHE     = (Noise_) ? 18 : int(2000*eHighHE_);
+        x_min   = (Noise_) ? -3.   : 0.;
+        x_max   = (Noise_) ? 3.    :  2.*eHighHE_;
 	for (int eta = -50; eta < 50; eta++) {
 	  for (int phi = 0; phi < 100; phi++) {
-	    for (int depth = 1; depth < maxDepthHE; depth++) {
+	    for (int depth = 1; depth <= maxDepthHE; depth++) {
 	      HcalDetId cell (HcalEndcap, eta, phi, depth);
 	      if (hcaltopology->valid(cell)) {
 		sprintf (name, "HEeta%dphi%ddep%d", eta, phi, depth);
-		sprintf (title,"Energy (HE #eta %d #phi %d depth %d)", eta, phi, depth);
-		TH1D* h = new TH1D(name, title, nbinHE, 0, 2*eHighHE_);
-		histHE_[cell] = h;
+		sprintf (title,"HE #eta %d #phi %d depth %d", eta, phi, depth);
+		TH1D* h = fs_->make<TH1D>(name, title, nbinHE, x_min, x_max);
+		histHC_[cell] = h;
 	      }
 	    }
 	  }
 	}
 	// For HF
 	int maxDepthHF = 4;
-	int nbinHF     = int(200*eHighHF_);
+	int nbinHF     = (Noise_) ? 200 : int(2000*eHighHF_);
+	x_min   = (Noise_) ? -10.   : 0.;
+	x_max   = (Noise_) ? 10.    :  2.*eHighHF_;
 	for (int eta = -50; eta < 50; eta++) {
 	  for (int phi = 0; phi < 100; phi++) {
-	    for (int depth = 1; depth < maxDepthHF; depth++) {
+	    for (int depth = 1; depth <= maxDepthHF; depth++) {
 	      HcalDetId cell (HcalForward, eta, phi, depth);
 	      if (hcaltopology->valid(cell)) {
 		sprintf (name, "HFeta%dphi%ddep%d", eta, phi, depth);
 		sprintf (title,"Energy (HF #eta %d #phi %d depth %d)", eta, phi, depth);
-		TH1D* h = new TH1D(name, title, nbinHF, 0, 2*eHighHF_);
-		histHF_[cell] = h;
+		TH1D* h = fs_->make<TH1D>(name, title, nbinHF, x_min, x_max);
+		histHC_[cell] = h;
```

Comment:
I see still some magic numbers around... If they can be derived from the topology of the different partitions, it would be better. But I let you figure out this.